tests/unit-core/test-api-instaceof.c

```diff
@@ -0,0 +1,104 @@
+/* Copyright JS Foundation and other contributors, http://js.foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "jerryscript.h"
+
+#include "test-common.h"
+
+static jerry_value_t
+my_constructor (const jerry_value_t func_val, /**< function */
+                const jerry_value_t this_val, /**< this */
+                const jerry_value_t argv[], /**< arguments */
+                const jerry_length_t argc) /**< number of arguments */
+{
+  (void) func_val;
+  (void) this_val;
+  (void) argv;
+  (void) argc;
+  return jerry_create_undefined ();
+} /* my_constructor */
+
+int
+main (void)
```

Comment:
There should be more test cases, for example:
* string, string
* string, number
* array, array
* array, object
* object, array
* null, object
* object, string
* typedarray, object
* arraybuffer, object
* symbols, object